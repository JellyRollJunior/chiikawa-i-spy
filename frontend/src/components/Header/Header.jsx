import { IconWrapper } from '../IconWrapper/IconWrapper';
import { Link } from 'react-router';
import styles from './Header.module.css';
import shared from '../../styles/shared.module.css';

const Header = ({ children }) => {
  return (
    <header className={styles.header}>
      <IconWrapper>
        <Link to="/">
          <button className={`${styles.headerButton} ${shared.button}`}>
            <h1 className={styles.title}>I Spy Chiikawa!</h1>
          </button>
        </Link>
      </IconWrapper>
      <nav>
        {children}
      </nav>
    </header>
  );
};

export { Header };
