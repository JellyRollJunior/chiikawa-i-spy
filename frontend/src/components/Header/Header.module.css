.header {
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 100%;
  padding: 12px;
  background-color: var(--yellow);

  border-top: 2px solid var(--green);
  border-bottom: 3px solid var(--green);
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
}

.title {
  font-size: 1.5rem;
}

.headerButton {
  color: var(--dark-green);
  background-color: inherit;
  border: none;
}

.headerButton:hover {
  background-color: var(--light-green);
}