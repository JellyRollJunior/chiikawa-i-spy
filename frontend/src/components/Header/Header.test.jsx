import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Header } from './Header';

describe('Header component', () => {
  it('renders title, header element, and children elements', () => {
    const { container } = render(
      <BrowserRouter>
        <Header>
          <button>I am a child!</button>
        </Header>
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
