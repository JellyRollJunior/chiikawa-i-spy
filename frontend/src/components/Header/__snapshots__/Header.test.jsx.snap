// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders title, header element, and children elements 1`] = `
<div>
  <header
    class="_header_63c2ea"
  >
    <div
      class="_iconWrapper_5d4a96"
    >
      <img
        alt="Chiikawa character icon"
        src="/src/assets/characterIcons/kurimanju-icon.png"
        style="height: 38px; width: 38px; margin-right: 3px;"
      />
      <a
        data-discover="true"
        href="/"
      >
        <button
          class="_headerButton_63c2ea _button_5890fc"
        >
          <h1
            class="_title_63c2ea"
          >
            I Spy Chiikawa!
          </h1>
        </button>
      </a>
      <img
        alt="Chiikawa character icon"
        src="/src/assets/characterIcons/momonga-icon.png"
        style="height: 38px; width: 38px; margin-left: 3px;"
      />
    </div>
    <nav>
      <button>
        I am a child!
      </button>
    </nav>
  </header>
</div>
`;
